feathr_project/feathr/utils/_envvariableutil.py

```diff
@@ -5,39 +5,101 @@
 from azure.core.exceptions import ResourceNotFoundError
 
 class _EnvVaraibleUtil(object):
-    def __init__(self, config_path):
+    """A utility class to read config variables.
+    If use_env_vars set to False, `get_environment_variable_with_default` will not use os environment variables.
+    Note, `get_environment_variable` still uses os environment variables.
+    """
+
+    def __init__(self, config_path: str, use_env_vars: bool = True):
+        """Initialize the utility class.
+
+        Args:
+            config_path: Config file path.
+            use_env_vars (optional): Whether to use os environment variables instead of config file. Defaults to True.
+        """
         self.config_path = config_path
-        # Set to none first to avoid invalid reference
-        self.akv_name = None
-        self.akv_name = self.get_environment_variable_with_default( 'secrets', 'azure_key_vault', 'name')
+        self.use_env_vars = use_env_vars
+
+        self.akv_name = (
+            self._get_variable_from_env("secrets__azure_key_vault__name") or
```

Comment:
Should `self._get_variable_from_env("secrets__azure_key_vault__name")` respect use_env_vars flag?

Reply:
make sense. will change that.